### DISCUSSION TITLE
I made SIGILS, a web-based item pipe replacement and factory manager

### DISCUSSION THREAD

**Discussion by fechan** [cc-tweaked/CC-Tweaked#1998] [Show and tell]
SIGILS, the *Shack Industries Graphical Item Logistics Software*, is a system where you connect your in-game machines to a ComputerCraft computer with wired modems and create virtual item pipes using a web browser.

Basically I was playing a modpack and I got tired of having to organize all my pipes just to break them when I needed to change them, so being a programmer I decided that the solution to all things is software.

A [tutorial for the basic usage of SIGILS](https://fredchan.org/blog/how-to-use-sigils/) is available. I intended to make a video about it but without any external motivation, I stopped working on it when I stopped playing my modded playthrough. I think most of my playthrough was spent programming SIGILS, though.

Anyone brave enough to try SIGILS can report issues to the [GitHub issues page](https://github.com/fechan/SIGILS/issues). Maybe if people start using it, I'll be motivated to work on it again.

Here's a screenshot of a basic mass-smelting setup. If you drag any machine into another machine, you can combine them into super-machines and make groups of their slots that you can create pipes to and from. This one combines furnaces, blast furnaces, and smokers into one mega-furnace:

![image](https://github.com/user-attachments/assets/69273257-efc1-4074-800b-cfb79bfbf1e7)

And here's a screenshot of a section of my factory from last playthrough, which processes ores in my massive Integrated Dynamics chest monster, generates obsidian with Create and Integrated Dynamics, and runs my Immersive Engineering potato-to-biodiesel refinery.

![image](https://github.com/user-attachments/assets/6dd92fc6-ff5c-4b5b-bff9-64bfa4b5d481)

Some mods don't play nice with the ComputerCraft Inventory API, such as Mekanism, so as a workaround I attach hoppers and Create chutes stuff to the sides of Mekanism machines and connect the hoppers to the wired modems instead, which SIGILS pipes items between.

**Comment by SquidDev** [comment]
Oh, this is incredibly cool! I'm amazed how much polish you've put into the UI here — it's like an even flashier steve's factory manager.

**Comment by TrueAssass1n** [comment]
As someone who has just scratched the surface of programming (less than a year total), this is really inspiring!!! I'll be using this for my current modpack playthrough!

**Comment by Star-SgrA** [comment]
Not very good at using it yet.
But I made some pipe to supply the crafter.Very nice!!
